arviz_dashboard/forest.py

```diff
@@ -0,0 +1,217 @@
+            .posterior.data_vars.variables[self.data_variable][0][0]
+            .size
+            > 1
+        ):
+            coor_variables = list(
+                self.idatas_cmp[self.model].posterior.indexes["school"]
+            )
+        else:
+            coor_variables = [""]
+        self.param["coor_variable"].objects = coor_variables
+        if self.coor_variable not in coor_variables:
+            self.coor_variable = coor_variables[0]
+
+
+class ForestDashboard(ModelVar):
+    def __init__(self, idatas_cmp) -> None:
+        self.idatas_cmp = idatas_cmp
+        super().__init__(self.idatas_cmp)
+
+    def dashboard_forest(self):
+        # define the widgets
+        multi_select = pn.widgets.MultiSelect(
+            name="ModelSelect",
+            options=list(self.idatas_cmp.keys()),
+            value=["mA"],
+        )
+        thre_slider = pn.widgets.FloatSlider(
+            name="HDI Probability",
+            start=0,
+            end=1,
+            step=0.05,
+            value=0.7,
+            width=200,
+        )
+        truncate_checkbox = pn.widgets.Checkbox(name="Ridgeplot Truncate")
+        ridge_quant = pn.widgets.RangeSlider(
+            name="Ridgeplot Quantiles",
+            start=0,
+            end=1,
+            value=(0.25, 0.75),
+            step=0.01,
+            width=200,
+        )
+        op_slider = pn.widgets.FloatSlider(
+            name="Ridgeplot Overlap",
+            start=0,
+            end=1,
+            step=0.05,
+            value=0.7,
+            width=200,
+        )
+
+        rope_slider = pn.widgets.RangeSlider(
+            name="Rope Range",
+            start=-10,
+            end=10,
+            value=(2, 5),
+            step=1,
+            width=200,
+        )
+
+        # construct widget
+        @pn.depends(
+            multi_select.param.value,
+            thre_slider.param.value,
+            rope_slider.param.value,
+            self.param.data_variable,
+            self.param.coor_variable,
+        )
+        def get_forest_plot(
+            multi_select, thre_slider, rope_slider, 
+            data_variable, coor_variable
+        ):
+            # generate graph
+            data = []
+            for model_ in multi_select:
+                data.append(self.idatas_cmp[model_])
+            # add rope
+            rope = {}
+            school = {}
+            school["school"] = coor_variable
+            school["rope"] = rope_slider
+            rope[data_variable] = [school]
+            # print(rope)
+            forest_plt = az.plot_forest(
+                data,
+                model_names=multi_select,
+                rope=rope,
+                kind="forestplot",
+                hdi_prob=thre_slider,
+                backend="bokeh",
+                figsize=(9, 9),
+                show=False,
+                combined=True,
+                colors="cycle",
+            )
+            return forest_plt[0][0]
```

Comment:
magic numbers again (`[0][0]`)

Reply:
this is an interesting question, so the `forest_plot` is a array as well as `forest_plot[0]`. Right now, I change it to `forest_plot.base[0]`

Reply:
You are correct that it is a nested array. I think this is a feature from the matplotlib interface. You can leave a comment indicating what it is, and why you are indexing `[0][0]` to get the bokeh plot. I'm not sure `forest_plot.base[0]` is any clearer than `[0][0]`